### PR TITLE
Feature/binomial test

### PR DESCRIPTION
**Pull request by du-phan**
[ch56723](https://app.clubhouse.io/dataiku/story/56723/add-binomial-test)

### CODE REVIEW

File: webapps/model-drift-view/app.js [hunk 1]
```diff
@@ -75,7 +86,8 @@ function draw(data) {
             break;
         case "REGRESSION":
             d3.select("#fugacity_div").selectAll("div").remove();
-            d3.select("#kde_container_div").select("h3").remove();
+            d3.select("#fugacity_label").remove();
```
Comment (tdesfont) [inline webapps/model-drift-view/app.js:89]: Do you remove the kde_container_div as in the previous version ?

File: webapps/model-drift-view/app.js [hunk 2]
```diff
@@ -117,18 +129,29 @@ function json2table(json, classes) {
         return string.charAt(0).toUpperCase() + string.slice(1);
     }
 
+    /*
```
Comment (tdesfont) [inline webapps/model-drift-view/app.js:132]: Unused code

File: webapps/model-drift-view/app.js [hunk 3]
```diff
@@ -117,18 +129,29 @@ function json2table(json, classes) {
...
-        body += '<tr>';
+        //body += '<tr>';
         cols.map(function(colName) {
-            body += '<td align="middle">' + row[colName] + '</td>';
+            body += '<td>' + row[colName] + '</td>';
         });
         body += '</tr>';
     });
-    return `<div><table class="${classes}"><thead><tr>${header}</tr></thead><tbody>${body}</tbody></table></div>`;
+    //return `<div><table class="${classes}"><thead><tr>${header}</tr></thead><tbody>${body}</tbody></table></div>`;
```
Comment (tdesfont) [inline webapps/model-drift-view/app.js:152]: header is in body now right ?

File: webapps/model-drift-view/body copy.html [hunk 4]
```diff
@@ -0,0 +1,102 @@
+<link rel="stylesheet" href="/plugins/model-drift/resource/style.css" />
```
Comment (tdesfont) [inline webapps/model-drift-view/body copy.html:1]: Is this copy necessary ?

Comment (tdesfont) [reply webapps/model-drift-view/body copy.html:1]: body.html // `body copy.html`

File: webapps/model-drift-view/body.html [hunk 5]
```diff
@@ -2,99 +2,146 @@
...
+                      <th>Conclusion</th>
+                      <td id="binomial-conclusion"><span>&#8804;</span> 0.05 so drift detected</td>
+                    </tr>
+                  </tbody>
+                </table>
+            </div>
+            <div class="col explanation text-r">
+                <b>The hypothesis tested</b> is that there is no drift, in which case the expected drift model accuracy is 0.5 (datasets undistinguishable). The observed accuracy might deviate from this expectation and the Binomial test evaluates whether this deviation is statistically significant, modelling the number of correct predictions as a random variable drawn from a Binomial distribution.
+                <br>
+                The p-value is the probability to observe this particular accuracy (or larger) under the hypothesis of absent drift. If this probability is lower than the significance level (i.e. 5%), it’s then unlikely to be in the situation of absent drift: the hypothesis of no drift is rejected, triggering a drift detection. The significance level indicates the rate of falsely-detected drifts we are ready to accept from the test.
```
Comment (tdesfont) [inline webapps/model-drift-view/body.html:94]: This sentence is a bit complex: it’s then unlikely to be in the situation of absent drift:

File: webapps/model-drift-view/body.html [hunk 6]
```diff
@@ -2,99 +2,146 @@
...
+    </div>
+    <hr/>
+    <div class="container-fluid">
+        <div class="medium-title-sb" style="padding-bottom:16px">Model Information</div>
+        <div class="small-title-r" id="fugacity_label" style="padding-bottom:16px">Fugacity</div>
+        <div class="row" id="fugacity_div">
+            <div class="col fugacity_score_container">
+                <div id="fugacity-score"></div>
+            </div>
+            <div class="col explanation text-r"> <b>Fugacity</b> expresses the difference between the expected "ideal" data your model was trained on and the observed "real" data you are analyzing. We compare the proportion of samples predicted in each class when scoring on both the test and your input datasets.
```
Comment (tdesfont) [inline webapps/model-drift-view/body.html:106]: `scoring on both the test and your input dataset**s**`, should it be dataset ?

File: webapps/model-drift-view/app.js (hunk 1 above)
Comment (du-phan) [reply webapps/model-drift-view/app.js:89]: yep I made some modifications to the html structure

File: webapps/model-drift-view/app.js (hunk 3 above)
Comment (du-phan) [reply webapps/model-drift-view/app.js:152]: yes the header css format is not very nice so I replace it by body

File: webapps/model-drift-view/body copy.html (hunk 4 above)
Comment (du-phan) [reply webapps/model-drift-view/body copy.html:1]: ah my mistake, should not have pushed it

File: webapps/model-drift-view/app.js (hunk 2 above)
Comment (du-phan) [reply webapps/model-drift-view/app.js]: removed in 8865aca

File: webapps/model-drift-view/body copy.html (hunk 4 above)
Comment (du-phan) [reply webapps/model-drift-view/body copy.html]: removed in 8865aca

File: webapps/model-drift-view/body.html (hunk 6 above)
Comment (du-phan) [reply webapps/model-drift-view/body.html:106]: hmm I think english-wise it should be dataset**s**

File: webapps/model-drift-view/body.html (hunk 5 above)
Comment (du-phan) [reply webapps/model-drift-view/body.html:94]: I'm not sure to understand your comment lol

Comment (tdesfont) [reply webapps/model-drift-view/body.html:94]: negation of negation but ok I think it is because of the hypothesis wording

Comment (du-phan) [reply webapps/model-drift-view/body.html:94]: yes indeed, we think that when communicating p-value, being in the reservation side is always a good idea

**Pull request merged**